### ISSUE TITLE
Allow loading fulltable module from jsdelivr

### ISSUE THREAD

**Issue by tacman** [fullcalendar/fullcalendar#7474]
### Reduced Test Case

https://jsfiddle.net/tacman1123/917t4mas/2/

### Do you understand that if a reduced test case is not provided, we will intentionally delay triaging of your ticket?

- [X] I understand

### Which connector are you using (React/Angular/etc)?

No connector (vanilla JS)

### Bug Description

Symfony's new AssetMapper component allows loading javascript as easily as loading php files using composer.

However, just like composer loads from packagist.org, importmap loads from jsdelivr.

```bash
bin/console importmap:require fullcalendar -vv

10:53:33 INFO      [http_client] Request: "GET https://data.jsdelivr.com/v1/packages/npm/fullcalendar/resolved?specifier=%2A"
10:53:34 INFO      [http_client] Response: "200 https://data.jsdelivr.com/v1/packages/npm/fullcalendar/resolved?specifier=%2A"
10:53:34 INFO      [http_client] Request: "GET https://cdn.jsdelivr.net/npm/fullcalendar@6.1.9/+esm"
10:53:34 INFO      [http_client] Request: "GET https://data.jsdelivr.com/v1/packages/npm/fullcalendar@6.1.9/entrypoints"
10:53:34 INFO      [http_client] Response: "200 https://cdn.jsdelivr.net/npm/fullcalendar@6.1.9/+esm"
10:53:34 INFO      [http_client] Response: "200 https://data.jsdelivr.com/v1/packages/npm/fullcalendar@6.1.9/entrypoints"
```

And as discussed in #7472 , right now the fullcalendar module doesn't load from jsdelivr, only from skypack, which is unmaintained.

**Comment by acerix** [comment]
Thanks for the details, it seems this may be a bug causing it to build incorrectly on jsdelvr.  I get the same "Uncaught TypeError: Class constructor component cannot be invoked without 'new'" error with the basic example and the "esm.run" URL's they suggest:

https://codepen.io/acerix/pen/YzdRZxN?editors=1000

You could also build the ESM version and host it yourself as an alternative to jsdelvr.

The example in the docs should also be changed from skypack as it's not maintained.

**Comment by tacman** [comment]
I hope that you're able to fix this so that it installs on jsdelivr.

I'm not working on the project that needed it anymore, but I built a Symfony bundle around it that I quite liked and want to finish by the time Symfony 7 is released next month.

While I can't help at all with the development, I'm happy to test.

**Comment by arcanis** [comment]
I have the same issue with another ESM bundler, and it's quite difficult to figure out what happens. In my case it seems that DayGridView is passed to the `customGenerator` prop. Do you see a reason why this might happen?

**Comment by jordann93** [comment]
Hello @tacman ,

I use this library in a new symfony 7 project with AssetMapper so I have this error..
What do you suggest to have a working version or fullcalendar please ?

Thanks a lot !

**Comment by tacman** [comment]
I wasn't able to get it working with AssetMapper, but I'm hopeful it'll be fixed here at some point, because the current cdn isn't at all stable.

**Comment by arshaw** [comment]
After some debugging I've found that

https://cdn.jsdelivr.net/npm/@fullcalendar/core@6.1.10/+esm

and

https://cdn.jsdelivr.net/npm/@fullcalendar/core@6.1.10/index.js/+esm
(which is called from the daygrid plugin)

are not being correctly deduplicated, resulting in two conflicting version of the fullcalendar core code.

We have all plugins (including daygrid) referencing the core peer dependency via the index.js path, because we found that some old bundlers got confused in projects that intermingled cjs/esm packages. We will revisit this in the next major version.

Until there's only one URL for the core plugin (or until jsdelivr starts correctly deduping), I'm afraid the only solution is to explicitly define how all endpoints, both public and internal, resolve. Like this:

https://github.com/fullcalendar/fullcalendar/blob/main/tests/manual/importmap.html#L6-L14

I'll make a note in the docs.

**Comment by tacman** [comment]
Thanks, @arshaw .  Happy to test as soon as it hits jsdelivr.

**Comment by tacman** [comment]
> because we found that some old bundlers got confused in projects that intermingled cjs/esm

@arshaw since jsdelivr is now the de facto bundler for running npmjs modules in the browser, I would respectfully ask to modify the code to run with it rather then the older bundlers that got confused.

I'm not sure which one you're referring to, many of the [old bundlers seems to have gone away](https://www.reddit.com/r/node/comments/16c7lxr/cdn_that_bundles_npm_packages_for_browsers/).  

Thanks for all you do with the library.  I have an app (for a non-profit, not getting paid for it) that is crying for a calendar, and fullcalendar is _the_ best option, and the calendar-bundle makes it super easy for Symfony developers to integrate.  But now that Symfony uses jsdelivr for its front-end assets, I'm stuck.

Would you consider making a fix for a minor release, instead of waiting for the next major release?  Or making this a major release (for BC) with really just this change.  fullcalendar is one of the most popular js libraries out there, and to have it working with jsdelivr would be awesome.

Alternatively, can you open an issue on jsdelivr and maybe the developer there can figure out the de-dupping?

Thanks again.

**Comment by tacman** [comment]
This is such a great package!  But not being supported on jsdelivr means I can no longer use it.  Please forgive the gentle nudges, I know it's a volunteer project for you, as are the projects I'm using it with.

**Comment by suguanYang** [comment]
I have the same issue, solved by external all `@fullcalendar/core/internal.js`  imports, make they all import the same module

**Comment by tacman** [comment]
@suguanYang Can you describe that in more detail, or cut and paste the code snippet?

**Comment by suguanYang** [comment]
Ok, I really mean I ran into simillar situation when I bundless all my dependencies and imported as ESM at runtime, I had to patch on @fullcalendar/core and @fullcalendar/timegrid's node_module and make sure they import the same @fullcalendar/core/internal-common.js module
and it just worked, so I guess there must have some module level variables that need to be shared between the 2 modules @fullcalendar/core and @fullcalendar/timegrid

then config the bundler make the @fullcalendar/core/internal-common.js externalize

may be @fullcalendar/core/internal-common.js should be extracted from @fullcalendar/core and published as a seperated package

this is my workaround, and it likely not suits for you
https://github.com/suguanYang/fullcalendar-esm/commit/0f8deca0fdb319613e906d3bd367d591e6c11b43

**Comment by tacman** [comment]
Thanks, @suguanYang , indeed, a bit more than I can handle.

**Comment by arshaw** [comment]
I tried debugging this further, but got stuck again. Seems that even if I take care of fullcalendar package deduplication, preact duplicates as well.

However, I have a working proof of concept where one must verbosely specify all internal/external paths from jsdelivr. Please check it out:
https://github.com/fullcalendar/fullcalendar/blob/main/tests/manual/importmap.public.html

**Comment by tacman** [comment]
@arshaw Kudos for putting that together!  

If the authors of this library had a funding amount to make this publishable on npm, perhaps we could raise the money.

**Comment by pfpro** [comment]
I spent some time today migrating a symfony project from Webpack to Assetmapper, got FullCalendar working via Assetmapper, and Claude Code led me to the following insights – perhaps they will be helpful.

Still experiencing this with @fullcalendar/core@6.1.20 and Symfony's AssetMapper (which uses jsdelivr as its CDN backend).     

**The Problem (two layers):**
                                                                                                                                
1. jsdelivr's +esm endpoint breaks FullCalendar — The Rollup+Terser processing causes the class constructor TypeError. This means importmap:require @fullcalendar/core (Symfony's standard way to add JS packages) doesn't work out of the box.            
2. Workaround creates friction — Using the raw npm ESM files directly works, but these files use relative imports (./internal-common.js). In an importmap-based setup (no bundler), every relative cross-file import needs a corresponding importmap entry, and updating requires sed-rewriting imports in 3 files after every download.                                  

**Current workaround:**

- Download raw npm files (index.js, internal.js, preact.js, internal-common.js) into a local directory
- Add explicit importmap entries for each file
- sed-rewrite ./internal-common.js → @fullcalendar/core/internal-common.js in 3 core files
- Repeat on every version update

**What would help (in order of impact):**

1. Fix the jsdelivr compatibility — If the +esm endpoint worked, none of the manual steps would be needed. This might require coordinating with jsdelivr or adjusting the build output to survive Rollup+Terser processing.
2. Publish importmap-ready ESM files — A build variant that uses bare specifiers (@fullcalendar/core/internal-common.js) instead of relative imports (./internal-common.js) would work natively with browser importmaps without any post-processing.
3. Reduce cross-file imports — Bundling internal-common.js into index.js would eliminate the most problematic relative import entirely.

This affects all frameworks using native importmaps without a bundler (Symfony AssetMapper, Rails 7 importmaps, etc.).

My importmap.php

```
<?php

/**
 * Returns the importmap for this application.
 *
 * - "path" is a path inside the asset mapper system. Use the
 *     "debug:asset-map" command to see the full list of paths.
 *
 * - "entrypoint" (JavaScript only) set to true for any module that will
 *     be used as an "entrypoint" (and passed to the importmap() Twig function).
 *
 * The "importmap:require" command can be used to add new entries to this file.
 */
return [
    ...
    'fullcalendar' => [
        'path' => './assets/scripts/global/fullcalendar.js',
        'entrypoint' => true,
    ],
    // FullCalendar core (raw npm in assets/lib/ — NOT managed by importmap:update)
    '@fullcalendar/core' => [
        'path' => './assets/lib/@fullcalendar/core/core.index.js',
    ],
    '@fullcalendar/core/index.js' => [
        'path' => './assets/lib/@fullcalendar/core/core.index.js',
    ],
    '@fullcalendar/core/internal.js' => [
        'path' => './assets/lib/@fullcalendar/core/internal.js',
    ],
    '@fullcalendar/core/preact.js' => [
        'path' => './assets/lib/@fullcalendar/core/preact.js',
    ],
    '@fullcalendar/core/internal-common.js' => [
        'path' => './assets/lib/@fullcalendar/core/internal-common.js',
    ],
    // FullCalendar locales (bare specifier needed by fullcalendar.js entrypoint)
    '@fullcalendar/core/locales/de.js' => [
        'path' => './assets/lib/@fullcalendar/core/locales/de.js',
    ],
    // Preact (raw npm in assets/lib/)
    'preact' => [
        'path' => './assets/lib/preact/preact.index.js',
    ],
    'preact/compat' => [
        'path' => './assets/lib/preact/compat.js',
    ],
    'preact/hooks' => [
        'path' => './assets/lib/preact/hooks.js',
    ],
    // FullCalendar plugins (version-managed — importmap:update works normally)
    '@fullcalendar/daygrid' => [
        'version' => '6.1.20',
    ],
    '@fullcalendar/timegrid' => [
        'version' => '6.1.20',
    ],
    '@fullcalendar/list' => [
        'version' => '6.1.20',
    ],
    '@fullcalendar/daygrid/internal.js' => [
        'version' => '6.1.20',
    ],
    '@fullcalendar/timegrid/internal' => [
        'version' => '6.1.20',
    ],
    '@fullcalendar/list/internal' => [
        'version' => '6.1.20',
    ],
];

```

**Comment by vince83110** [comment]
Thanks @pfpro for your solution, since you seem very involved, do you know if version 7 would fix anything about this ? 

Or should we try to allow Symfony importmap to use esm.sh  / skypack for example ? I use skypack now, but hardcoded, I don't like this solution.

**Comment by tacman** [comment]
When AssetMapper was under development, I posted a message to the jsDelivr support (github issues, I think), and the developer got back to me pretty quickly and I think even tweaked something for a problematic package.

If you post an issue, he may have some insights as to how to approach this.

Gut feeling is that somebody's going to fork this library, vibe-code a solution, and republish it.  And that the new version will be more robust not just for ESM but for doing forward.

**Comment by pfpro** [comment]
## Summary

FullCalendar v7 (currently beta.8 / rc.0) introduces native importmap support, making it significantly better suited for our Symfony AssetMapper setup.

## Key Changes in v7

### Improvements
- **Single package**: `fullcalendar` replaces all `@fullcalendar/*` packages, with subpath exports (`fullcalendar/daygrid`, `fullcalendar/timegrid`, etc.)
- **Official importmap support**: FullCalendar now ships a working importmap reference example and actively tests browser-native imports
- **Clean ESM exports**: No more `internal.js` / `internal-common.js` spaghetti — proper ESM with `"type": "module"`
- **New headless core**: `@full-ui/headless-calendar` extracted as framework-agnostic kernel

### Unchanged / New Challenges
- **Preact still required** (5 importmap entries)
- **New dependencies**: `@full-ui/headless-calendar` + `temporal-polyfill` (3 subpath entries for TC39 Temporal API)
- **More importmap entries overall**: ~16 vs ~10-12 currently
- **New CSS theming system**: `skeleton.css` + theme files (classic, monarch, breezy, etc.) — completely different from v6

**Comment by vince83110** [comment]
Skypack is totally dead now, since few days, lot of errors.

I tried to use 7.0.0-beta.8 with `importmap:require`, but get the error:

```
installHook.js:1 Error connecting controller

TypeError: Class constructor component cannot be invoked without 'new'
    at Un.render (fullcalendar.index-rCLNzRL.js:7:40841)
    at N (preact.index-h1OtZFJ.js:7:6437)
```

With importmap.php: 

```
'@fullcalendar/core/locales/fr' => [
        'version' => '6.1.20',
    ],
    'fullcalendar' => [
        'version' => '7.0.0-beta.8',
    ],
    'preact/jsx-runtime' => [
        'version' => '10.23.2',
    ],
    'preact/compat/client' => [
        'version' => '10.23.2',
    ],
    '@full-ui/headless-calendar' => [
        'version' => '7.0.0-beta.8',
    ],
    'temporal-polyfill/fns/zoneddatetime' => [
        'version' => '0.3.2',
    ],
    'temporal-polyfill/fns/plaindatetime' => [
        'version' => '0.3.2',
    ],
    'temporal-polyfill/fns/instant' => [
        'version' => '0.3.2',
    ],
    'fullcalendar/daygrid' => [
        'version' => '7.0.0-beta.8',
    ],
    'fullcalendar/timegrid' => [
        'version' => '7.0.0-beta.8',
    ],
    'fullcalendar/interaction' => [
        'version' => '7.0.0-beta.8',
    ],
```

So gonna wait and use the solution from pfpro

**Comment by arshaw** [comment]
here's a working importmap example that works with v7-beta.8:

https://github.com/fullcalendar/fullcalendar-workspace/blob/v7-dev/standard/packages/vanilla-tests/manual/importmap.public.html

**Comment by clussiana** [comment]
I'm still facing this issue when using v7 (rc.2) with Asset Mapper
Indeed @pfpro workaround works, but it's not optimal

**Comment by tacman** [comment]
Confirming this and, hopefully, closing the loop for anyone arriving here from Symfony **AssetMapper** / native import maps.

**It's the jsDelivr `/+esm` build, not FullCalendar.** AssetMapper's import-map resolver downloads the `https://cdn.jsdelivr.net/npm/<pkg>/+esm` variant of each package. That `+esm` build of FullCalendar throws at runtime; the **raw published ESM of the identical version works**.

Minimal reproduction — **no bundler, no Symfony**, just three `+esm` imports in a plain HTML page:

```html
<script type="module">
  const { Calendar } = await import('https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/+esm');
  const daygrid = (await import('https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/daygrid/+esm')).default;
  new Calendar(document.getElementById('cal'), { plugins:[daygrid], initialView:'dayGridMonth' }).render();
</script>
```
→
```
TypeError: Class constructor component cannot be invoked without 'new'
    at Tn.render (https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/+esm:7:37050)
    at $ (https://cdn.jsdelivr.net/npm/preact@10.29.1/+esm …)   ← preact diff
```

The crash is inside FullCalendar's render, where a generator is dispatched with `typeof r === "function" ? r(props, h) : r` — under the `+esm` build a preact `Component` **class** reaches that branch and is invoked without `new`. Swap the two URLs to the **raw** files (`…/fullcalendar@7.0.0-rc.3/index.js`, `…/daygrid.js`) and it renders.

I checked the obvious suspects: it is **not** preact duplication (at a single version, `(await import('…/preact@10.29.1/+esm')).Component === (await import('…/preact@10.29.1/compat/+esm')).Component` is `true`), and FullCalendar's `+esm` uses native `class extends` (no ES5 downlevel). Something in how jsDelivr's `+esm` bundles FullCalendar's class components is the trigger — I've filed a parallel issue with jsDelivr. Reproduces with **both v6 and v7** via `/+esm`.

**The fix: use the raw published ESM, not `/+esm`.** FullCalendar's own importmap manual test (`standard/packages/vanilla-tests/manual/importmap.public.html`) already does exactly this and works. The raw files load their own `./chunks/*` from the CDN and resolve bare specifiers (`preact`, `fullcalendar`, …) through the import map. Verified rendering (v7.0.0-rc.3) under Symfony AssetMapper with this import map:

```html
<script type="importmap">
{
  "imports": {
    "preact": "https://cdn.jsdelivr.net/npm/preact@10.29.1/dist/preact.mjs",
    "preact/hooks": "https://cdn.jsdelivr.net/npm/preact@10.19.5/hooks/dist/hooks.mjs",
    "preact/jsx-runtime": "https://cdn.jsdelivr.net/npm/preact@10.19.5/jsx-runtime/dist/jsxRuntime.mjs",
    "preact/compat": "https://cdn.jsdelivr.net/npm/preact@10.19.5/compat/dist/compat.mjs",
    "preact/compat/client": "https://cdn.jsdelivr.net/npm/preact@10.19.5/compat/client.mjs",
    "temporal-polyfill/fns/zoneddatetime": "https://cdn.jsdelivr.net/npm/temporal-polyfill@0.3.2/fns/zoneddatetime.js",
    "temporal-polyfill/fns/plaindatetime": "https://cdn.jsdelivr.net/npm/temporal-polyfill@0.3.2/fns/plaindatetime.js",
    "temporal-polyfill/fns/instant": "https://cdn.jsdelivr.net/npm/temporal-polyfill@0.3.2/fns/instant.js",
    "@full-ui/headless-calendar": "https://cdn.jsdelivr.net/npm/@full-ui/headless-calendar@7.0.0-rc.3/index.js",
    "fullcalendar": "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/index.js",
    "fullcalendar/themes/classic": "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/themes/classic.js",
    "fullcalendar/daygrid": "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/daygrid.js",
    "fullcalendar/timegrid": "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/timegrid.js",
    "fullcalendar/list": "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.0-rc.3/list.js"
  }
}
</script>
```

CSS in v7 is separate (no JS auto-injection): `…/fullcalendar@7.0.0-rc.3/skeleton.css` + `themes/classic/theme.css` + `themes/classic/palette.css`.

For Symfony specifically, the practical workaround is to **remove** `fullcalendar`/`preact` from `importmap.php` (so AssetMapper doesn't fetch `/+esm`) and supply the raw URLs in a second `<script type="importmap">` (browsers merge multiple import maps). I've filed a corresponding request on the Symfony side so AssetMapper can do this without the manual step.

TL;DR: **`/+esm` mangles FullCalendar; the raw published ESM works.** Might be worth a docs note for import-map / AssetMapper / Vite-CDN users.

**Comment by wzrdjstr** [comment]
## Summary of workarounds

[EsmShEsmResolver.php](https://github.com/user-attachments/files/29477984/EsmShEsmResolver.php)

### 1. Run fullcalendar v5, which doesn't have the bundling issue.

**NOTE:** v5 is two major versions behind the latest version.

https://github.com/fullcalendar/fullcalendar/issues/7472#issuecomment-2094293045


### 2. Use a CDN. Earlier comments recommended Skypack, which is no longer maintained. 
[esm.sh](https://esm.sh) appears to work as a drop-in replacement for Skypack (e.g. `import {Calendar} from 'https://esm.sh/@fullcalendar/core@6.1.15'`)

**NOTE:** If CDN goes down, your site may go down too.

https://github.com/fullcalendar/fullcalendar/issues/7472#issuecomment-1748636080
https://github.com/fullcalendar/fullcalendar/issues/7472#issuecomment-2331236904


### 3. Use a bundler (e.g. [rolldown](https://rolldown.rs/)) to build and host your own fullcalendar bundle.
e.g. generate bundle inside `assets/components/fullcalendar` and import from this bundle in your Stimulus controllers.

**NOTE:** Adds a bundling step, which somewhat defeats the purpose of asset mapper.

https://github.com/fullcalendar/fullcalendar/issues/7474#issuecomment-1749781054


### 4. Add `fullcalendar` to your production `package.json` and configure asset mapper to serve files out of your `node_modules` directory using [local npm packages](https://symfony.com/doc/current/frontend/asset_mapper.html#using-local-npm-packages).

**NOTE:** Need to serve files from `node_modules` directory in production.


### 5. Run `importmap:require` command and explicitly require fullcalendar's internal/external packages.

https://github.com/fullcalendar/fullcalendar/issues/7474#issuecomment-2226318239
https://github.com/fullcalendar/fullcalendar/blob/main/tests/manual/importmap.public.html

https://github.com/fullcalendar/fullcalendar/issues/7474#issuecomment-1953317254
https://github.com/fullcalendar/fullcalendar/blob/main/tests/manual/importmap.html#L6-L14

**NOTE:** Requires knowledge of fullcalendar package structure that will need to be manually kept in sync.


### 6. Run `importmap:require` command for the usual packages and then patch fullcalendar source code.

https://github.com/fullcalendar/fullcalendar/issues/7474#issuecomment-2214634931
https://github.com/fullcalendar/fullcalendar/issues/7474#issuecomment-3993819323

**NOTE:** Need to apply and maintain patches.


### 7. Run alternate package resolver

As a workaround, I wrote a package resolver (attached) based on Symfony's [`EsmJsDelivrResolver`](https://github.com/symfony/symfony/blob/3750e07b661710271fa48f1139d6f58711558f15/src/Symfony/Component/AssetMapper/ImportMap/Resolver/JsDelivrEsmResolver.php#L26). This resolver uses [esm.sh](https://esm.sh) instead of [jsDelivr](https://www.jsdelivr.com/). Depending on your situation, this resolver might be a good stopgap until the jsDelivr issue is fixed.

Ideally we'd be able to configure a resolver on a per-package basis (e.g. https://github.com/symfony/symfony/issues/53999). In the meantime, you'll need to completely switch resolvers via `services.yaml` config:

```yaml
asset_mapper.importmap.resolver:
    class: App\Symfony\EsmShEsmResolver
    arguments:
        $target: es2015 # defaults to es2022
```

**Note:** The `importmap:update` command fails for FullCalendar v7 because of the way the latest package version is resolved. IIRC the `importmap:update` command just grabs whatever default version the CDN has pinned to the package name. Since the [Esm.sh fullcalendar](https://esm.sh/fullcalendar) package currently points to v6, it throws exceptions when trying to update v7 packages. The `importmap:update` command should start working again when Esm.sh points to v7. As a workaround, I've been manually updating the importmap version property of affected packages.

**Comment by tacman** [comment]
I think the jsdelivr issues is fixed, as I've been using this in a regular, Symfony 7 project with no special handling.

**Comment by vince83110** [comment]
@tacman I tried this morning, with the stable version 7.0.1 and Symfony Asset Mapper, still the error: 

```
installHook.js:1 Error connecting controller

TypeError: Class constructor nl cannot be invoked without 'new'
    at Ve.render (fullcalendar.index-5KYflsU.js:7:39662)
    at K (preact.index-9RaWEuF.js:7:6125)
    at le (preact.index-9RaWEuF.js:7:1838)
    at K (preact.index-9RaWEuF.js:7:6501)
    at le (preact.index-9RaWEuF.js:7:1838)
    at K (preact.index-9RaWEuF.js:7:6501)
    at le (preact.index-9RaWEuF.js:7:1838)
    at K (preact.index-9RaWEuF.js:7:6501)
    at le (preact.index-9RaWEuF.js:7:1838)
    at we (preact.index-9RaWEuF.js:7:8483)
```

Did you success to have it working ? Can you please share your importmap ? 
Thanks

**Comment by tacman** [comment]
https://github.com/survos/ux-calendar-bundle

Here's working example:

https://ccal.survos.com/

The importmap goes through something to cache the assets, these are the raw settings.

```php
<?php

/**
 * Returns the importmap for this application.
 *
 * - "path" is a path inside the asset mapper system. Use the
 *     "debug:asset-map" command to see the full list of paths.
 *
 * - "entrypoint" (JavaScript only) set to true for any module that will
 *     be used as an "entrypoint" (and passed to the importmap() Twig function).
 *
 * The "importmap:require" command can be used to add new entries to this file.
 *
 * @return array<string, array{    // Import name as key, description of the imported file as value
 *     path: string,               // Logical, relative or absolute path to the file
 *     type?: 'js'|'css'|'json',   // Type of the file, defaults to 'js'
 *     entrypoint?: bool,          // Whether the file is an entrypoint, for 'js' only
 * }|array{
 *     version: string,            // Version of the remote package
 *     package_specifier?: string, // Remote "package-name/path" specifier, defaults to the import name
 *     type?: 'js'|'css'|'json',
 *     entrypoint?: bool,
 * }>
 */
return [
    'app' => ['path' => './assets/app.js', 'entrypoint' => true],
    '@hotwired/stimulus' => ['version' => '3.2.2'],
    'bootstrap' => ['version' => '5.3.3'],
    '@popperjs/core' => ['version' => '2.11.8'],
    'bootstrap/dist/css/bootstrap.min.css' => ['version' => '5.3.3', 'type' => 'css'],
    '@symfony/stimulus-bundle' => ['path' => './vendor/symfony/stimulus-bundle/assets/dist/loader.js'],
    'twig' => ['version' => '1.17.1'],
    'locutus/php/strings/sprintf' => ['version' => '2.0.16'],
    'locutus/php/strings/vsprintf' => ['version' => '2.0.16'],
    'locutus/php/math/round' => ['version' => '2.0.16'],
    'locutus/php/math/max' => ['version' => '2.0.16'],
    'locutus/php/math/min' => ['version' => '2.0.16'],
    'locutus/php/strings/strip_tags' => ['version' => '2.0.16'],
    'locutus/php/datetime/strtotime' => ['version' => '2.0.16'],
    'locutus/php/datetime/date' => ['version' => '2.0.16'],
    'locutus/php/var/boolval' => ['version' => '2.0.16'],
    'axios' => ['version' => '1.6.7'],
    'fos-routing' => ['version' => '0.0.6'],
    'perfect-scrollbar' => ['version' => '1.5.5'],
    'perfect-scrollbar/css/perfect-scrollbar.min.css' => ['version' => '1.5.5', 'type' => 'css'],
    'datatables.net-plugins/i18n/en-GB.mjs' => ['version' => '2.0.8'],
    'datatables.net-bs5' => ['version' => '2.1.3'],
    'jquery' => ['version' => '3.7.1'],
    'datatables.net' => ['version' => '2.1.3'],
    'datatables.net-bs5/css/dataTables.bootstrap5.min.css' => ['version' => '2.1.3', 'type' => 'css'],
    'datatables.net-buttons-bs5' => ['version' => '3.1.1'],
    'datatables.net-buttons' => ['version' => '3.1.1'],
    'datatables.net-buttons-bs5/css/buttons.bootstrap5.min.css' => ['version' => '3.1.1', 'type' => 'css'],
    'datatables.net-responsive-bs5' => ['version' => '3.0.2'],
    'datatables.net-responsive' => ['version' => '3.0.2'],
    'datatables.net-responsive-bs5/css/responsive.bootstrap5.min.css' => ['version' => '3.0.2', 'type' => 'css'],
    'datatables.net-scroller-bs5' => ['version' => '2.4.3'],
    'datatables.net-scroller' => ['version' => '2.4.3'],
    'datatables.net-scroller-bs5/css/scroller.bootstrap5.min.css' => ['version' => '2.4.3', 'type' => 'css'],
    'datatables.net-searchpanes-bs5' => ['version' => '2.3.0'],
    'datatables.net-searchpanes' => ['version' => '2.3.0'],
    'datatables.net-searchpanes-bs5/css/searchPanes.bootstrap5.min.css' => ['version' => '2.3.0', 'type' => 'css'],
    'datatables.net-searchbuilder-bs5' => ['version' => '1.7.0'],
    'datatables.net-searchbuilder' => ['version' => '1.7.0'],
    'datatables.net-searchbuilder-bs5/css/searchBuilder.bootstrap5.min.css' => ['version' => '1.7.0', 'type' => 'css'],
    'datatables.net-select-bs5' => ['version' => '2.0.4'],
    'datatables.net-select' => ['version' => '2.0.4'],
    'datatables.net-select-bs5/css/select.bootstrap5.min.css' => ['version' => '2.0.4', 'type' => 'css'],
    'preact' => ['version' => '10.19.5'],
    'preact/hooks' => ['version' => '10.19.5'],
    'preact/jsx-runtime' => ['version' => '10.19.5'],
    'preact/compat' => ['version' => '10.19.5'],
    'preact/compat/client' => ['version' => '10.19.5'],
    'temporal-polyfill/fns/zoneddatetime' => ['version' => '0.3.2'],
    'temporal-polyfill/fns/plaindatetime' => ['version' => '0.3.2'],
    'temporal-polyfill/fns/instant' => ['version' => '0.3.2'],
    '@full-ui/headless-calendar' => ['version' => '7.0.0-beta.8'],
    'fullcalendar' => ['version' => '7.0.0-beta.8'],
    'fullcalendar/themes/classic' => ['version' => '7.0.0-beta.8'],
    'fullcalendar/interaction' => ['version' => '7.0.0-beta.8'],
    'fullcalendar/daygrid' => ['version' => '7.0.0-beta.8'],
    'fullcalendar/timegrid' => ['version' => '7.0.0-beta.8'],
    'fullcalendar/list' => ['version' => '7.0.0-beta.8'],
];

```

Source is at https://github.com/survos/ccal

**Comment by wzrdjstr** [comment]
@vince83110 I still have this issue too; using FullCalendar 7.0.1 and Symfony 8.1. 

I'm currently using the CDN workaround (since my `EsmShEsmResolver.php` now produces the same error):

```js
// HACK: Import from CDN until jsdelivr issue has been resolved
// https://github.com/fullcalendar/fullcalendar/issues/7474
import { Calendar } from "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.1/index.js";
import multiMonthPlugin from "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.1/multimonth.js";
import listPlugin from "https://cdn.jsdelivr.net/npm/fullcalendar@7.0.1/list.js";
// NOTE: https://cdn.jsdelivr.net/npm/fullcalendar@7.0.1/themes/breezy.js returns 404
import themePlugin from "https://cdn.jsdelivr.net/npm/fullcalendar/themes/breezy.js";

// import { Calendar } from "fullcalendar";
// import multiMonthPlugin from "fullcalendar/multimonth";
// import listPlugin from "fullcalendar/list";
// import themePlugin from "fullcalendar/themes/breezy";

import "fullcalendar/skeleton.css";
import "fullcalendar/themes/breezy/theme.css";
import "fullcalendar/themes/breezy/palettes/indigo.css";
```

My `importmap.php` looks like this:

```php
return [
    ...
    'fullcalendar' => ['version' => '7.0.1'],
    'fullcalendar/themes/breezy' => ['version' => '7.0.1'],
    'fullcalendar/skeleton.css' => ['version' => '7.0.1', 'type' => 'css'],
    'fullcalendar/skeleton.min.css' => ['version' => '7.0.1', 'type' => 'css'],
    'fullcalendar/themes/breezy/palettes/indigo.css' => ['version' => '7.0.1', 'type' => 'css'],
    'fullcalendar/themes/breezy/theme.css' => ['version' => '7.0.1', 'type' => 'css'],
    'fullcalendar/multimonth' => ['version' => '7.0.1'],
    'fullcalendar/list' => ['version' => '7.0.1'],
    'preact/jsx-runtime' => ['version' => '10.29.7'],
    'preact/compat' => ['version' => '10.29.7'],
    'preact/compat/client' => ['version' => '10.29.7'],
    '@full-ui/headless-calendar' => ['version' => '7.0.1'],
    'preact' => ['version' => '10.29.7'],
    'preact/hooks' => ['version' => '10.29.7'],
    'temporal-polyfill/fns/ZonedDateTime' => ['version' => '1.0.1'],
    'temporal-polyfill/fns/PlainDateTime' => ['version' => '1.0.1'],
    'temporal-polyfill/fns/Instant' => ['version' => '1.0.1'],
    'temporal-utils' => ['version' => '1.0.1'],
    'temporal-utils/protected' => ['version' => '1.0.1'],
    'temporal-utils/protected-error-messages' => ['version' => '1.0.1'],
    ...
]
```

**Comment by arshaw** [comment]
I feel like these giant maps like in the above comment should be part of the published distributable, as a json file

**Comment by vince83110** [comment]
Any updates on this ? Did anyone try with 7.1 ? Thanks in advance. I tried, but still... Just maybe I missed something.
